### ISSUE TITLE
Smoother transition for "notfound_image" for bolt 3.2 -> 3.3

### ISSUE THREAD

**Issue by bobdenotter** [bolt/bolt#6648]
If you update a site from bolt 3.2 to bolt 3.3, thumbnails will no longer work. 

![screen shot 2017-05-10 at 17 11 50](https://cloud.githubusercontent.com/assets/1833361/25905998/c68e9cd8-35a3-11e7-82f0-451442206cfe.png)

Right-click them, and you'll see the error: 

![screen shot 2017-05-10 at 17 09 28](https://cloud.githubusercontent.com/assets/1833361/25905975/b431e68a-35a3-11e7-8d85-8b73d236de0c.png)

The cause is this: 

```
thumbnails:
    ...
    notfound_image: view/img/default_notfound.png
    error_image: view/img/default_error.png
```

which needs to be updated to: 

```
    notfound_image: bolt_assets://img/default_notfound.png
    error_image: bolt_assets://img/default_error.png
```

Seeing how we have tons of people coming in, who we need to tell "Right click the thumbnail, read the error and install the module it's telling you about", I can see that this is going to cause a lot of headaches when we release 3.3

I suggest we change the code to make this transition a bit smoother.. Perhaps something like this, in `function findDefaultImage()`: 

```
        // Take old-style (pre 3.3) paths into account:
        // view/img/foo.png => bolt_assets://img/foo.png
        if (strpos($path, '://') === false) {
            $path = str_replace('view/', 'bolt_assets://', $path);
        }
```

btw: If an image exists, why does the thumbnailer even need to read the `notfound_image` ?

**Comment by CarsonF** [comment]
Yes this is correct. There is code that is _supposed_ to handle the old value, but it's clearly not working. I'll try to see what's up with that code.

Another thing we could do is update the value on composer update. With the PR I have in coming there will be a nice spot for that.



> btw: If an image exists, why does the thumbnailer even need to read the notfound_image ?

It just needs to pass it along. The existence check might be able to be delayed until it's needed but I would have to look deeper.

**Comment by bobdenotter** [comment]
Sounds good! 

> Another thing we could do is update the value on composer update.

Will that work, if the enduser has modified the default value?

> It just needs to pass it along.

Sure, but isn't that a performance hit? Or, it's probably really, really minor, and if you're bothered by it, you should turn on `save_files: true`, right? ;-)

**Comment by CarsonF** [comment]
> Will that work, if the enduser has modified the default value?

No, but in that case we could give a message explaining the change maybe. Hopefully the code change will handle BC in that case.

> Sure, but isn't that a performance hit? Or, it's probably really, really minor, and if you're bothered by it, you should turn on save_files: true, right? ;-)

It's really minor. I don't think `save_files` has anything to do with that though.

**Comment by bobdenotter** [comment]
> No, but in that case we could give a message explaining the change maybe. 

I could easily add a message for that in `bolt/configuration-helpers`. :-)

> It's really minor. I don't think save_files has anything to do with that though.

Sure it does. If it's on, the file will be saved as-is to the filesystem, and following requests will bypass PHP entirely and just serve the static file.

**Comment by CarsonF** [comment]
Happy if you want to merge that, but in general I think we need a better way (in core) to detect deprecated config keys and values.

**Comment by bobdenotter** [comment]
@CarsonF It's not "one or the other". We should absolutely have what you suggest, but _also_ the user-friendly notification.

**Comment by GawainLynch** [comment]
> … but in general I think we need a better way (in core) to detect deprecated config keys and values.

You mean like the validation in the config branch I just picked up again yesterday? :wink:

**Comment by bobdenotter** [comment]
Closed by #6728
